src/tools/perf/lib/ucp_tests.cc

```diff
@@ -1044,24 +1044,62 @@ ucs_status_t ucp_perf_test_dispatch(ucx_perf_context_t *perf)
         (UCX_PERF_CMD_FADD,  UCX_PERF_TEST_TYPE_STREAM_UNI),
         (UCX_PERF_CMD_SWAP,  UCX_PERF_TEST_TYPE_STREAM_UNI),
         (UCX_PERF_CMD_CSWAP, UCX_PERF_TEST_TYPE_STREAM_UNI)
-        );
+    );
+    return UCS_ERR_INVALID_PARAM;
+}
 
+static ucs_status_t dispatch_tag(ucx_perf_context_t *perf)
+{
     UCS_PP_FOREACH(TEST_CASE_ALL_TAG, perf,
         (UCX_PERF_CMD_TAG,      UCX_PERF_TEST_TYPE_PINGPONG),
         (UCX_PERF_CMD_TAG,      UCX_PERF_TEST_TYPE_STREAM_UNI),
         (UCX_PERF_CMD_TAG_SYNC, UCX_PERF_TEST_TYPE_PINGPONG),
         (UCX_PERF_CMD_TAG_SYNC, UCX_PERF_TEST_TYPE_STREAM_UNI)
-        );
+    );
+    return UCS_ERR_INVALID_PARAM;
+}
 
+static ucs_status_t dispatch_stream(ucx_perf_context_t *perf)
+{
     UCS_PP_FOREACH(TEST_CASE_ALL_STREAM, perf,
         (UCX_PERF_CMD_STREAM,   UCX_PERF_TEST_TYPE_STREAM_UNI),
         (UCX_PERF_CMD_STREAM,   UCX_PERF_TEST_TYPE_PINGPONG)
-        );
+    );
+    return UCS_ERR_INVALID_PARAM;
+}
 
+static ucs_status_t dispatch_am(ucx_perf_context_t *perf)
+{
     UCS_PP_FOREACH(TEST_CASE_ALL_AM, perf,
         (UCX_PERF_CMD_AM,       UCX_PERF_TEST_TYPE_PINGPONG),
         (UCX_PERF_CMD_AM,       UCX_PERF_TEST_TYPE_STREAM_UNI)
-        );
+    );
+    return UCS_ERR_INVALID_PARAM;
+}
+
+ucs_status_t ucp_perf_test_dispatch(ucx_perf_context_t *perf)
+{
+    ucs_status_t status;
+
+    status = dispatch_osd(perf);
```

Comment:
maybe replace by an array of function pointers

Reply:
thank you, done

Reply:
Outdated, done in other PRs